aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/UpdateHandler.java

```diff
@@ -58,7 +58,11 @@ protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
             final VersionedProxyClient<RdsClient> rdsProxyClient,
             final VersionedProxyClient<Ec2Client> ec2ProxyClient
     ) {
-        if (!ImmutabilityHelper.isChangeMutable(request.getPreviousResourceState(), request.getDesiredResourceState())) {
+        final ProxyClient<RdsClient> rdsClient = rdsProxyClient.defaultClient();
+
+        final DBInstance instance = StringUtils.isNullOrEmpty(request.getPreviousResourceState().getEngine()) ?
+                fetchDBInstance(rdsClient, request.getPreviousResourceState()) : null;
```

Comment:
In case when the customer has out-of bounds deleted the instance, this code will produce InternalError due to NPE. Can we surround fetchDBInstance with try-catch clause or add it to call-chain invocation with error handling to make sure we return NotFound error?